msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerAccountManagerStrategy.java

```diff
@@ -56,397 +48,305 @@
-        final List<ICacheRecord> cacheRecords = new ArrayList<>();
-        Logger.verbose(
-                TAG + methodName,
-                "Retrieve all the accounts from account manager with broker account type, "
-                        + "and the account length is: " + accountList.length
-        );
+        if (!BrokerMsalController.isAccountManagerPermissionsGranted(parameters.getAppContext())) {
+            //If the account manager permissions are not granted, return false.
+            return false;
+        }
 
-        if (accountList == null || accountList.length == 0) {
-            Telemetry.emit(
-                    new BrokerEndEvent()
-                            .putAction(methodName)
-                            .isSuccessful(false)
-                            .putErrorCode(ErrorStrings.NO_ACCOUNT_FOUND)
-            );
-            return cacheRecords;
-        } else {
-            final Bundle bundle = new Bundle();
-            bundle.putBoolean(DATA_CACHE_RECORD, true);
-            bundle.putInt(AuthenticationConstants.Broker.CALLER_INFO_UID,
-                    Binder.getCallingUid());
-            bundle.putString(ACCOUNT_CLIENTID_KEY, parameters.getClientId());
-
-            for (final Account eachAccount : accountList) {
-                // Use AccountManager Api method to get extended user info
-
-                final AccountManagerFuture<Bundle> result = AccountManager.get(parameters.getAppContext())
-                        .updateCredentials(
-                                eachAccount,
-                                AuthenticationConstants.Broker.AUTHTOKEN_TYPE,
-                                bundle,
-                                null,
-                                null,
-                                null
-                        );
-
-                final Bundle userInfoBundle = result.getResult();
-                cacheRecords.addAll(
-                        MsalBrokerResultAdapter.accountsFromBundle(userInfoBundle)
-                );
+        return invokeBrokerAccountManagerOperation(parameters, new OperationInfo<OperationParameters, Boolean>() {
+            @Override
+            public Bundle getRequestBundle(OperationParameters parameters) {
+                final Bundle requestBundle = mRequestAdapter.getRequestBundleForHello(parameters);
+                requestBundle.putString(AuthenticationConstants.Broker.BROKER_ACCOUNT_MANAGER_OPERATION_KEY,
+                        AuthenticationConstants.BrokerAccountManagerOperation.HELLO);
+                return requestBundle;
             }
 
-            Telemetry.emit(
-                    new BrokerEndEvent()
-                            .putAction(methodName)
-                            .isSuccessful(true)
-            );
+            @Override
+            public String getMethodName() {
+                return ":helloWithAccountManager";
```

Comment:
This does not match the actual method name, is that intended ?

Reply:
Yes.